src/material/core/tokens/m2/mdc/_switch.scss

```diff
@@ -0,0 +1,182 @@
+  $on-surface-state-content: if($is-dark, #fafafa, #212121);
+  $disabled-handle-color: if($is-dark, #000, #424242);
+  $icon-color: if($is-dark, #212121, #fff);
+
+  // The default for tokens that rely on the theme will use the primary palette
+  $theme-color-tokens: get-color-palette-color-tokens(
+    map.get($config, primary),
+    $is-dark
+  );
+
+  @return map.merge(
+    $theme-color-tokens,
+    (
+      // Color of handle when selected while disabled.
+      disabled-selected-handle-color: $disabled-handle-color,
+      // Color of handle when selected while disabled.
+      disabled-selected-icon-color: $icon-color,
+      // Color of track when selected while disabled.
+      disabled-selected-track-color: $on-surface,
+      // Color of handle when unselected while disabled.
+      disabled-unselected-handle-color: $disabled-handle-color,
+      // Color of icon when unselected while disabled.
+      disabled-unselected-icon-color: $icon-color,
+      // Color of track when disabled.
+      disabled-unselected-track-color: $on-surface,
+      // Color of slide-toggle handle's surface.
+      handle-surface-color: surface,
+      // Elevation level for handle.
+      handle-elevation: 1,
+      // Color of handle's shadow
+      handle-shadow-color: if($elevation != null, $elevation, elevation.$color),
+      // Elevation level for handle when disabled.
+      disabled-handle-elevation: 0,
+      // Color of icon (ex. checkmark) when selected
+      selected-icon-color: $icon-color,
+      // Color of handle when unselected and focused.
+      unselected-focus-handle-color: $on-surface-state-content,
+      // Color of ripple when unselected and focused.
+      unselected-focus-state-layer-color: $on-surface,
+      // Color of track when unselected and focused.
+      unselected-focus-track-color: $hairline,
+      // Color of handle when unselected.
+      unselected-handle-color: $on-surface-variant,
+      // Color of handle when unselected and on hover.
+      unselected-hover-handle-color: $on-surface-state-content,
+      // Color of ripple when unselected and on hover.
+      unselected-hover-state-layer-color: $on-surface,
+      // Color of track when unselected and on hover.
+      unselected-hover-track-color: $hairline,
+      // Color of icon color when unselected.
+      unselected-icon-color: $icon-color,
+      // Color of handle when unselected and pressed.
+      unselected-pressed-handle-color: $on-surface-state-content,
+      // Color of ripple when unselected and pressed.
+      unselected-pressed-state-layer-color: $on-surface,
+      // Color of track when unselected and pressed.
+      unselected-pressed-track-color: $hairline,
+      // Color of track when selected.
+      unselected-track-color: $hairline,
+    )
+  );
+}
+
+// Generates the mapping for the properties that change based on the slide toggle color.
+@function get-color-palette-color-tokens($color-palette, $is-dark) {
```

Comment:
Might be a good idea to prefix the name with `private-` so it doesn't look like this a public API.